TFNetworkRecLayer.py

```diff
@@ -3268,20 +3268,29 @@ def __init__(self, beam_size, input_type="prob", explicit_search_source=None, le
         scores_in = tf.log(scores_in)
       elif input_type == "log_prob":
         pass
+      elif input_type == "regression":
+        pass
       else:
         raise Exception("%r: invalid input type %r" % (self, input_type))
-      samples = tf.multinomial(scores_in, num_samples=1, seed=get_random_seed())  # (batch, num_samples), int64
-      samples = tf.to_int32(tf.reshape(samples, [-1]))  # (batch,), int32
-      if self.scheduled_sampling.get("gold_mixin_prob"):
+
+      if input_type is not "regression":
+        samples = tf.multinomial(scores_in, num_samples=1, seed=get_random_seed())  # (batch, num_samples), int64
+        samples = tf.to_int32(tf.reshape(samples, [-1]))  # (batch,), int32
+        if self.scheduled_sampling.get("gold_mixin_prob"):
+          gold_targets = self._get_target_value().get_placeholder_as_batch_major()  # (batch,), int32
+          samples = tf.where(
+            tf.less(tf.random_uniform(tf.shape(samples)), self.scheduled_sampling.get("gold_mixin_prob")),
+            gold_targets, samples)
+      else:
         gold_targets = self._get_target_value().get_placeholder_as_batch_major()  # (batch,), int32
-        samples = tf.where(
-          tf.less(tf.random_uniform(tf.shape(samples)), self.scheduled_sampling.get("gold_mixin_prob")),
-          gold_targets, samples)
+        choice = tf.distributions.Bernoulli(probs=self.scheduled_sampling.get("gold_mixin_prob"),dtype=tf.bool).sample(sample_shape=tf.shape(scores_in)[0])
+        samples = tf.where(choice, gold_targets, scores_in)
```

Comment:
This code violates PEP8. (Spacing.)
Btw, isn't it possible to share the same code with the code above, for `samples`?
Also, this code should only run if `self.scheduled_sampling.get("gold_mixin_prob")` is actually set (like in the code above).

Reply:
The only line that could be shared is the gold_targets = ... line, I think there is no convenient way to share it

Reply:
Also the `tf.where`, or not?